lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -2918,6 +2918,55 @@ FuncInfo* PostVisitFunction(ParseNode* pnode, ByteCodeGenerator* byteCodeGenerat
 
     AssignFuncSymRegister(pnode, byteCodeGenerator, top);
 
+    if (pnode->sxFnc.pnodeBody && pnode->sxFnc.HasReferenceableBuiltInArguments() && pnode->sxFnc.UsesArguments() &&
+        pnode->sxFnc.HasHeapArguments())
+    {
+        bool doStackArgsOpt = top->byteCodeFunction->GetDoBackendArgumentsOptimization();
+                
+        bool hasAnyParamInClosure = false;
+
+        if (top->GetHasLocalInClosure())
+        {
+            ParseNodePtr currentFormalNode = pnode->sxFnc.pnodeParams;
+
+            while (currentFormalNode != nullptr)
+            {
+                if (currentFormalNode->sxPid.sym->GetHasNonLocalReference())
+                {
+                    hasAnyParamInClosure = true;
+                    break;
+                }
+
+                if (currentFormalNode->sxParamPattern.pnodeNext == nullptr)
+                {
+                    currentFormalNode = nullptr;
+                }
+                else
+                {
+                    currentFormalNode = currentFormalNode->sxParamPattern.pnodeNext;
+                }
+            }
```

Comment:
Can't we just do paramScope->GetHasOwnLocalInClosure() here instead of going through the list? In merge case also we are not resetting this flag.